swagger/swagger-generator/generator-core/src/test/java/org/apache/servicecomb/swagger/generator/core/TestParamUtils.java

```diff
@@ -95,4 +105,49 @@ public void setParameterTypeByTypeOnComplexType() {
           + "param name is [testName]", e.getMessage());
     }
   }
+
+  private static class AllTypeTest1 {
+    TestType1 t1;
+
+    List<TestType1> t2;
+
+    Map<String, TestType1> t3;
+
+    TestType1[] t4;
+  }
+
+  private static class AllTypeTest2 {
+    TestType2 t1;
+
+    List<TestType2> t2;
+
+    Map<String, TestType2> t3;
+
+    TestType2[] t4;
+  }
+
+  @Test
+  public void testAddDefinitions() {
+    Field[] fields1 = AllTypeTest1.class.getDeclaredFields();
+    Field[] fields2 = AllTypeTest2.class.getDeclaredFields();
+    for (int i = 0; i < fields1.length; i++) {
+      for (int j = 0; j < fields2.length; j++) {
+        if(fields1[i].isSynthetic()||fields2[j].isSynthetic()){
```

Comment:
jacoco will use ASM injecting bytecode into field,this caused a new field named JacocoData will not throw exception when execute  addDefinitions.